Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -115,80 +115,158 @@ public void Roslyn_Settings_ValidSetup()
+            };
 
-            // Act
-            var result = BuildRunner.BuildTargets(TestContext, projectRoot.FullPath, TargetConstants.OverrideRoslynAnalysisTarget);
+            var testSpecificProjectXml = @"
+  <PropertyGroup>
+    <ResolvedCodeAnalysisRuleSet>c:\original.ruleset</ResolvedCodeAnalysisRuleSet>
+    <Language>C#</Language>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- all analyzers specified in the project file should be preserved -->
+    <Analyzer Include='c:\original.analyzer1.dll' />
+    <Analyzer Include='original.analyzer2.dll' />
+  </ItemGroup>
+  <ItemGroup>
+    <!-- These additional files don't match ones in the config and should be preserved -->
+    <AdditionalFiles Include='should.not.be.removed.additional1.txt' />
+    <AdditionalFiles Include='should.not.be.removed.additional2.txt' />
+
+    <!-- This additional file matches one in the config and should be replaced -->
+    <AdditionalFiles Include='should.be.removed\CONFIG.1.TXT' />
+
+  </ItemGroup>
+";
+            var projectFilePath = CreateProjectFile(config, testSpecificProjectXml);
 
-            var projectSpecificConfFilePath = result.GetCapturedPropertyValue(TargetProperties.ProjectConfFilePath);
-            var expectedRoslynAdditionalFiles = new string[] { projectSpecificConfFilePath }
-                .Concat(analyzerAdditionalFiles)
-                .Concat(notRemovedAdditionalFiles)
-                .ToArray();
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, projectFilePath, TargetConstants.OverrideRoslynAnalysisTarget);
 
             // Assert
+            result.AssertTargetExecuted(TargetConstants.CreateProjectSpecificDirs);
             result.AssertTargetExecuted(TargetConstants.OverrideRoslynAnalysisTarget);
             result.AssertTargetExecuted(TargetConstants.SetRoslynAnalysisPropertiesTarget);
             result.BuildSucceeded.Should().BeTrue();
 
             // Check the error log and ruleset properties are set
             AssertErrorLogIsSetBySonarQubeTargets(result);
-            AssertExpectedResolvedRuleset(result, "d:\\my.ruleset");
-            AssertExpectedItemValuesExists(result, TargetProperties.AdditionalFilesItemType, expectedRoslynAdditionalFiles);
-            AssertExpectedItemValuesExists(result, TargetProperties.AnalyzerItemType, expectedAssemblies);
+
+            var actualProjectSpecificOutFolder = result.GetCapturedPropertyValue(TargetProperties.ProjectSpecificOutDir);
+            Directory.Exists(actualProjectSpecificOutFolder).Should().BeTrue();
+
+            var expectedMergedRuleSetFilePath = Path.Combine(actualProjectSpecificOutFolder, "merged.ruleset");
+            AssertExpectedResolvedRuleset(result, expectedMergedRuleSetFilePath);
+            RuleSetAssertions.CheckMergedRulesetFile(actualProjectSpecificOutFolder,
+                @"c:\original.ruleset", "d:\\generated.ruleset");
+
+
+            // TODO - processing of additional files and analyzer
```

Comment:
TODO - in next commits. So far the test is just checking that the merged ruleset was created correctly.